lib/integrations/externalcloudaudit/connection.go

```diff
@@ -0,0 +1,173 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package externalcloudaudit
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/athena"
+	athenatypes "github.com/aws/aws-sdk-go-v2/service/athena/types"
+	"github.com/aws/aws-sdk-go-v2/service/glue"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	ecatypes "github.com/gravitational/teleport/api/types/externalcloudaudit"
+	"github.com/gravitational/trace"
+)
+
+const (
+	dummyFileName = "_connection-test"
+	//empty string md5 base64 encoded
+	contentMD5                  = "1B2M2Y8AsgTpgAmY7PhCfg=="
+	queryExecutionTimeout       = time.Minute
+	queryExecutionInterval      = 100 * time.Millisecond
+	queryExecutionIntervalDelay = 300 * time.Millisecond
+)
+
+// ConnectionTestAthenaClient is a subset of [athena.Client] methods needed for athena test connection.
+type ConnectionTestAthenaClient interface {
+	// Runs the SQL query statements contained in the Query
+	StartQueryExecution(ctx context.Context, params *athena.StartQueryExecutionInput, optFns ...func(*athena.Options)) (*athena.StartQueryExecutionOutput, error)
+	// Returns information about a single execution of a query if you have access to the workgroup in which the query ran.
+	GetQueryExecution(ctx context.Context, params *athena.GetQueryExecutionInput, optFns ...func(*athena.Options)) (*athena.GetQueryExecutionOutput, error)
+	// Streams the results of a single query execution specified by QueryExecutionId from the Athena query results location in Amazon S3.
+	GetQueryResults(ctx context.Context, params *athena.GetQueryResultsInput, optFns ...func(*athena.Options)) (*athena.GetQueryResultsOutput, error)
+}
+
+// ConnectionTestGlueClient is a subset of [glue.Client] methods needed for athena test connection.
+type ConnectionTestGlueClient interface {
+	// Retrieves the Table definition in a Data Catalog for a specified table.
+	GetTable(ctx context.Context, params *glue.GetTableInput, optFns ...func(*glue.Options)) (*glue.GetTableOutput, error)
+}
+
+// ConnectionTestS3Client is a subset of [s3.Client] methods needed for athena test connection.
+type ConnectionTestS3Client interface {
+	// Adds an object to a bucket.
+	PutObject(ctx context.Context, params *s3.PutObjectInput, optFns ...func(*s3.Options)) (*s3.PutObjectOutput, error)
+}
+
+func ConnectionTestBuckets(ctx context.Context, clt ConnectionTestS3Client, spec *ecatypes.ExternalCloudAuditSpec) error {
```

Comment:
If we move the call site for these functions to the gRPC service in teleport.e, might as well move this code to teleport.e as well. I'd suggest `e/lib/externalcloudaudit/connection.go`